toolchain/driver/runtimes_cache_test.cpp

```diff
@@ -0,0 +1,717 @@
+  result = Error("nothing");
+  EXPECT_TRUE(*tmp_dir_.Access("test_home"));
+  EXPECT_TRUE(*test_home.Access(".cache"));
+
+  // Same as with an empty string, but with a relative path instead.
+  setenv(XdgCacheEnv, "relative/cache/home", /*overwrite*/ true);
+  result = Runtimes::Cache::MakeSystem(install_);
+  ASSERT_THAT(result, IsSuccess(_));
+  EXPECT_THAT(result->path(), StartsWith(home_cache_path));
+  // Destruction shouldn't remove system cache directories.
+  result = Error("nothing");
+  EXPECT_TRUE(*tmp_dir_.Access("test_home"));
+  EXPECT_TRUE(*test_home.Access(".cache"));
+
+  // Same as with an empty string, but this time with an unset environment
+  // variable.
+  unsetenv(XdgCacheEnv);
+  result = Runtimes::Cache::MakeSystem(install_);
+  ASSERT_THAT(result, IsSuccess(_));
+  EXPECT_THAT(result->path(), StartsWith(home_cache_path));
+  // Destruction shouldn't remove system cache directories.
+  result = Error("nothing");
+  EXPECT_TRUE(*tmp_dir_.Access("test_home"));
+  EXPECT_TRUE(*test_home.Access(".cache"));
+
+  // Now check a bunch of different failure modes for the home directory
+  // fallback. These should all end up creating temporary directories which
+  // we'll test functionally at the end.
+  setenv(HomeEnv, "", /*overwrite*/ true);
+  EXPECT_THAT(Runtimes::Cache::MakeSystem(install_), IsSuccess(_));
+  setenv(HomeEnv, "relative/home", /*overwrite*/ true);
+  EXPECT_THAT(Runtimes::Cache::MakeSystem(install_), IsSuccess(_));
+
+  // Correct the path and make sure it works again.
+  setenv(HomeEnv, home_path.c_str(), /*overwrite*/ true);
+  result = Runtimes::Cache::MakeSystem(install_);
+  ASSERT_THAT(result, IsSuccess(_));
+  EXPECT_THAT(result->path(), StartsWith(home_cache_path));
+
+  // Now try removing directories around home.
+  ASSERT_THAT(test_home.Rmtree(".cache"), IsSuccess(_));
+  EXPECT_THAT(Runtimes::Cache::MakeSystem(install_), IsSuccess(_));
+  ASSERT_THAT(tmp_dir_.Rmtree("test_home"), IsSuccess(_));
+  EXPECT_THAT(Runtimes::Cache::MakeSystem(install_), IsSuccess(_));
+
+  // Finally, double check that these temporary caches still produce a writable
+  // directory.
+  result = Runtimes::Cache::MakeSystem(install_);
+  ASSERT_THAT(result, IsSuccess(_));
+  EXPECT_THAT(result->path(), Not(StartsWith(home_cache_path)));
+  EXPECT_THAT(result->path(), Not(StartsWith(xdg_path)));
+  ASSERT_THAT(Filesystem::Cwd().WriteFileFromString(
+                  result->path() / "test_file", "test"),
+              IsSuccess(_));
+  ASSERT_THAT(Filesystem::Cwd().ReadFileToString(result->path() / "test_file"),
+              IsSuccess(StrEq("test")));
+}
+
+TEST_F(RuntimesCacheTest, BasicBuild) {
+  llvm::SmallVector<std::string> targets = {"aarch64-unknown-unknown",
+                                            "x86_64-unknown-unknown"};
+  llvm::SmallVector<std::filesystem::path> built_runtimes_paths;
+
+  for (const std::string& target : targets) {
+    auto lookup_result = cache_.Lookup({.target = target});
```

Comment:
```suggestion
    SCOPED_TRACE(target);
    auto lookup_result = cache_.Lookup({.target = target});
```

(So that failure messages tell you which iteration the failure happened on)